Nos/Models/NoteParser.swift

```diff
@@ -15,34 +21,32 @@ struct NoteParser {
     /// Parses the content and tags stored in a note and returns an attributed text with tagged entities replaced
     /// with readable names.
     func parse(content: String, tags: [[String]], context: NSManagedObjectContext) -> AttributedString {
-        var result = replaceTaggedNostrEntities(in: content, tags: tags, context: context)
-        result = replaceNostrEntities(in: result)
-        do {
-            return try AttributedString(
-                markdown: result,
-                options: AttributedString.MarkdownParsingOptions(interpretedSyntax: .inlineOnlyPreservingWhitespace)
-            )
-        } catch {
-            return AttributedString(stringLiteral: content)
-        }
+        let replaced = replaceTaggedNostrEntities(in: content, tags: tags, context: context)
+        let (result, _) = replaceNostrEntities(in: replaced)
+        return (try? AttributedString(
+            markdown: result,
+            options: AttributedString.MarkdownParsingOptions(interpretedSyntax: .inlineOnlyPreservingWhitespace)
+        )) ?? AttributedString(content)
     }
-
-    /// Parses the content and tags stored in a note and returns an attributed string and list of URLs that can be used 
+    
+    /// Parses the content and tags stored in a note and returns components that can be used
     /// to display the note in the UI.
-    func parse(content: String, tags: [[String]], context: NSManagedObjectContext) -> (AttributedString, [URL]) {
+    func parse(content: String, tags: [[String]], context: NSManagedObjectContext) -> NoteDisplayComponents {
```

Comment:
I think it's time to rename this or the other `parse(content:tags:context:)` -- it's too confusing and annoying to have to specify the return type when we call either of them. I know you didn't name it, but would you mind fixing it? I'd suggest `components(...)` like Apple's Calendar API.

```suggestion
    func components(from content: String, tags: [[String]], context: NSManagedObjectContext) -> NoteDisplayComponents {
```